### PR TITLE
Deprecates and replaces the Gravatar helper with a simplified version…

### PR DESCRIPTION
**Pull request by gsteel**
|    Q          |   A
|-------------- | ------
| Documentation | yes/no
| Bugfix        | yes/no
| BC Break      | yes/no
| New Feature   | yes/no
| RFC           | yes/no
| QA            | yes/no

### Description

The current Gravatar view helper has a tonne of setters and getters, dodgy inheritance and checks out superglobals to figure out if the current http request is secure or not.

The existing helper is deprecated completely in this pull with a replacement implementation that simply spits out an `img` tag with the relevant `src`.

Notable differences…
- Uses protocol relative url for the `src` to avoid inspecting the request
- Closes the img tag with /> regardless of whether the current doc is XHTML/HTML5

Because this was originally part of #146 - it imports the psalm type `AttributeSet` from HtmlAttributesSet which is added in #146 so psalm qa will fail.

Maybe this pull can be left for when/if 146 is merged and then rebased so psalm passes.

### CODE REVIEW

**Comment by Ocramius** [comment]
Needs another skim before review: CI failing 😁

File: src/Helper/GravatarImage.php [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+use Laminas\Escaper\Escaper;
+use Laminas\View\HtmlAttributesSet;
+
+use function md5;
+use function sprintf;
+use function strtolower;
+use function trim;
+
+/**
+ * @psalm-import-type AttributeSet from HtmlAttributesSet
```
Comment (Ocramius) [inline src/Helper/GravatarImage.php:16]: Needs a rebase to re-trigger CI :)

Comment (gsteel) [reply src/Helper/GravatarImage.php:16]: Rebased :)

**Review by Ocramius — APPROVED** [review]
LGTM 👍

**Comment by froschdesign** [comment]
@gsteel 
Updates for documentation are missing:

* deprecation notice for the [current helper](https://docs.laminas.dev/laminas-view/helpers/gravatar/) is missing
* the new helper must be documented
* [migration guide](https://github.com/laminas/laminas-view/pull/122) must be extended

**Comment by Ocramius** [comment]
@froschdesign I would leave all notices completely out, for stuff that can be deprecated statically. Notices are only necessary for combinations of input parameters.

**Comment by froschdesign** [comment]
@Ocramius I mean the notice in the documentation: https://docs.laminas.dev/laminas-view/helpers/gravatar/

**Comment by gsteel** [comment]
Sorry @froschdesign - #158 is ready for review :)

Would you be able to add recently merged pulls to the v3 project under done? This will help me when it comes to going over the deprecations/removals for the migration guide.

**Pull request merged**